src/backend/cuda/CMakeLists.txt

```diff
@@ -314,6 +316,47 @@ CL_KERNEL_TO_H(
     EOF "1"
     )
 
+SET(libdevice_bc "")
+IF (USE_LIBDEVICE)
+  SET(libdevice_computes "")
+  LIST(APPEND libdevice_computes "20" "30" "35" "50")
+  FOREACH(libdevice_compute ${libdevice_computes})
+    SET(_libdevice_bc_file "${CUDA_NVVM_HOME}/libdevice/libdevice.compute_${libdevice_compute}.10.bc")
+    SET(_libdevice_bc_copy "${CMAKE_BINARY_DIR}/src/backend/cuda/compute_${libdevice_compute}.bc")
+    IF (EXISTS ${_libdevice_bc_file})
+      ADD_CUSTOM_COMMAND(
+        OUTPUT "${_libdevice_bc_copy}"
+        DEPENDS "${_libdevice_bc_file}"
+        COMMAND ${CMAKE_COMMAND} -E copy "${_libdevice_bc_file}" "${_libdevice_bc_copy}")
+      LIST(APPEND libdevice_bc ${_libdevice_bc_copy})
+      ADD_DEFINITIONS(-D"__LIBDEVICE_COMPUTE_${libdevice_compute}")
+    ENDIF()
+  ENDFOREACH()
+ENDIF()
+
+LIST(LENGTH libdevice_bc libdevice_bc_len)
+
+IF (${libdevice_bc_len} GREATER 0)
+
+  SET(libdevice_headers
+    "libdevice_headers")
+
+  CL_KERNEL_TO_H(
+    SOURCES ${libdevice_bc}
+    VARNAME libdevice_files
+    EXTENSION "hpp"
+    OUTPUT_DIR ${libdevice_headers}
+    TARGETS libdevice_targets
+    NAMESPACE "cuda"
+    EOF "1"
+    )
+
+  MESSAGE(STATUS "LIBDEVICE found.")
+  ADD_DEFINITIONS(-DUSE_LIBDEVICE)
+ELSE()
+  MESSAGE(STATUS "LIBDEVICE not found on system. CUDA JIT may be slower")
```

Comment:
I don't think "CUDA JIT may be slower" succinctly talks about the performance. I think it may cause confusion over what "slower" is, which in fact is not actually slow, ie. there is no performance degradation compared to previous versions of ArrayFire (as I understand it at least).